internal/backend/s3/s3.go

```diff
@@ -87,6 +87,53 @@ func open(ctx context.Context, cfg Config, rt http.RoundTripper) (*Backend, erro
 		debug.Log("using anonymous access for %#v", cfg.Endpoint)
 	}
 
+	// assume role
+	roleArn := os.Getenv("AWS_ASSUME_ROLE_ARN")
+	if roleArn != "" {
+		sessionName := os.Getenv("AWS_ASSUME_ROLE_SESSION_NAME")
+		externalID := os.Getenv("AWS_ASSUME_ROLE_EXTERNAL_ID")
+		policy := os.Getenv("AWS_ASSUME_ROLE_POLICY")
+		stsEndpoint := os.Getenv("AWS_ASSUME_ROLE_STS_ENDPOINT")
+
+		if stsEndpoint == "" {
+			if len(os.Getenv("AWS_REGION")) > 0 {
+				if strings.HasPrefix(os.Getenv("AWS_REGION"), "cn-") {
+					stsEndpoint = "https://sts." + os.Getenv("AWS_REGION") + ".amazonaws.com.cn"
+				} else {
+					stsEndpoint = "https://sts." + os.Getenv("AWS_REGION") + ".amazonaws.com"
+				}
+			} else {
+				stsEndpoint = "https://sts.amazonaws.com"
+			}
+		}
+
+		opts := credentials.STSAssumeRoleOptions{
+			RoleARN:   roleArn,
+			AccessKey: c.AccessKeyID,
+			SecretKey: c.SecretAccessKey,
+		}
+		if c.SessionToken != "" {
+			opts.SessionToken = c.SessionToken
```

Comment:
As these options default to empty strings, there's no need for the surrounding if. This also allows inlining `os.Getenv("AWS_ASSUME_ROLE_SESSION_NAME")`, external Id and policy.

Reply:
Done.